src/components/Tree/Tree.cy.tsx

```diff
@@ -34,37 +66,231 @@ const TreeComponent = ({ onSelect }: { onSelect?: (id: Nullable<string>) => void
+                        <div>
+                            <button>Hello</button>
+                        </div>
+                    }
+                />
+            </Tree>,
+        );
+
+        cy.get(TREE_ITEM_ID).find('button').should('exist').contains('Hello');
+    });
+
+    it('should trigger onSelect as a controlled component when selecting and item', () => {
+        const onSelectStub = cy.stub().as('onSelectStub');
+
+        cy.mount(<TreeComponent onSelect={onSelectStub} />);
+
+        cy.get(TREE_ITEM_ID).first().click();
+
+        cy.get('@onSelectStub').should('have.been.calledOnce');
+    });
+
+    it('should trigger onSelect as a controlled component when deselecting and item', () => {
+        const onSelectStub = cy.stub().as('onSelectStub');
+
+        cy.mount(<TreeComponent onSelect={onSelectStub} selectedIds={['1']} />);
+
+        cy.get(TREE_ITEM_ID).first().click();
+
+        cy.get('@onSelectStub').should('have.been.calledOnce');
+    });
+
+    it('expands/shrinks the correct items as uncontrolled component', () => {
         cy.mount(<TreeComponent />);
 
-        cy.get(TREE_ITEM_ID).should('have.length', 3);
         cy.get(TREE_ITEM_TOGGLE_ID).first().click();
         cy.get(TREE_ITEM_ID).should('have.length', 8);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get(TREE_ITEM_ID).should('have.length', 3);
+    });
+
+    it('should trigger onExpand as a controlled component', () => {
+        const onExpandStub = cy.stub().as('onExpandStub');
+
+        cy.mount(<TreeComponent onExpand={onExpandStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click();
+        cy.get('@onExpandStub').should('have.been.calledOnce');
     });
 
-    it('renders all tree items', () => {
+    it('should trigger onShrink as a controlled component', () => {
+        const onShrinkStub = cy.stub().as('onShrinkStub');
+
+        cy.mount(<TreeComponent onShrink={onShrinkStub} />);
+
+        cy.get(TREE_ITEM_TOGGLE_ID).first().click().click();
+        cy.get('@onShrinkStub').should('have.been.calledOnce');
+    });
+
+    it('does not show then drag handle when not draggable', () => {
         cy.mount(<TreeComponent />);
 
-        cy.get(TREE_ITEM_TOGGLE_ID).eq(0).click();
-        cy.get(TREE_ITEM_TOGGLE_ID).eq(2).click();
-        cy.get(TREE_ITEM_TOGGLE_ID).eq(1).click();
+        cy.get(TREE_ITEM_ID).first().realHover();
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).should('not.be.visible');
+    });
+
+    it('shows then drag handle and allows dragging when draggable', () => {
+        cy.mount(<TreeComponent draggable />);
+
+        cy.get(TREE_ITEM_ID).first().realHover();
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).should('be.visible');
+        cy.get(TREE_ITEM_OVERLAY_ID).should('not.exist');
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).realMouseDown();
+        cy.get(TREE_ITEM_OVERLAY_ID).should('be.visible');
+        cy.get(TREE_ITEM_DRAG_HANDLE_ID).first().realMouseUp();
```

Comment:
Wouldn't it be nice to add `cy.get(TREE_ITEM_OVERLAY_ID).should('not.exist');` after the `realMouseUp()` just to make sure the drag handle disappears?